bug: `No records returned in stream` warning for ignored streams

### Singer SDK Version

0.45.1

### Is this a regression?

- [ ] Yes

### Python Version

3.9

### Bug scope

Taps (catalog, state, etc.)

### Operating System

_No response_

### Description

Ignoring no records for a stream surfaces another `UserWarning` I didn't previously spot after working on #2903, #2906 and #2907:

```
No records returned in stream
```

https://github.com/ReubenFrankel/tap-f1/actions/runs/13897554712

Should there a be a `UserWarning` emitted here if a developer is explicitly choosing to ignore no records for streams?

https://github.com/meltano/sdk/blob/c6672eb70002c7430f5db30fba05bb21cf9f0c11/singer_sdk/testing/tap_tests.py#L106:L111

### Link to Slack/Linen

_No response_

> Should there a be a `UserWarning` emitted here if a developer is explicitly choosing to ignore no records for streams?

I would say probably not.

Is this warning independent from the error in the linked workflow?

```
FAILED tests/test_core.py::TestTapF1::test_tap_stream_primary_keys[driver_standings] - KeyError: 'position'
```

--- a/singer_sdk/testing/tap_tests.py
+++ b/singer_sdk/testing/tap_tests.py
@@ -5,6 +5,7 @@ from __future__ import annotations
 import typing as t
 import warnings
 
+import pytest
 from jsonschema import validators
 from jsonschema.exceptions import SchemaError
 
@@ -107,11 +108,13 @@ class StreamReturnsRecordTest(StreamTestTemplate):
             self.config.ignore_no_records
             or self.stream.name in self.config.ignore_no_records_for_streams
         ):
-            # only warn if this or all streams are set to ignore no records
-            warnings.warn(UserWarning(no_records_message), stacklevel=2)
-        else:
-            record_count = len(self.stream_records)
-            assert record_count > 0, no_records_message
+            pytest.skip(
+                "Skipping test because no records were returned in "
+                f"stream '{self.stream.name}'",
+            )
+
+        record_count = len(self.stream_records)
+        assert record_count > 0, no_records_message
 
 
 class StreamCatalogSchemaMatchesRecordTest(StreamTestTemplate):

--- a/tests/samples/test_tap_csv.py
+++ b/tests/samples/test_tap_csv.py
@@ -75,11 +75,14 @@ _TestCSVOneStreamPerFileIncremental = get_tap_test_class(
         "delimiter": "\t",
     },
     state=STATE,
-    suite_config=SuiteConfig(ignore_no_records=True),
 )
 
 
 class TestCSVOneStreamPerFileIncremental(_TestCSVOneStreamPerFileIncremental):
+    @pytest.mark.xfail(
+        reason="There are no records because the state is set to the future.",
+        strict=True,
+    )
     def test_tap_stream_returns_record(
         self,
         config: SuiteConfig,
@@ -87,8 +90,16 @@ class TestCSVOneStreamPerFileIncremental(_TestCSVOneStreamPerFileIncremental):
         runner: TapTestRunner,
         stream: CSVStream,
     ):
-        with pytest.warns(
-            UserWarning,
-            match="No records returned in stream",
-        ):
-            super().test_tap_stream_returns_record(config, resource, runner, stream)
+        super().test_tap_stream_returns_record(config, resource, runner, stream)
+
+
+TestCSVOneStreamPerFileIncrementalIgnoreNoRecords = get_tap_test_class(
+    tap_class=SampleTapCSV,
+    config={
+        "path": "fixtures/csv",
+        "read_mode": "one_stream_per_file",
+        "delimiter": "\t",
+    },
+    state=STATE,
+    suite_config=SuiteConfig(ignore_no_records=True),
+)

